qutip/core/options.py

```diff
@@ -34,7 +34,8 @@ def __setitem__(self, key, value):
     def __repr__(self):
```

Comment:
@Ericgig Thanks for your advice! I added 'full option'. When full=True, `print(settings.compile.__repr__(full=True))` outputs everything. When full=False, it only outputs modified settings.

Reply:
Yes so now, `print(settings.compile)` will print the full output as expected. We still want to have `print(settings)` print the shorten version.

Also if you can, when no option is printed in the short version, it could print something like:
```
    ...
    colorblind_safe: False
    compile: <CompilationOptions()>
```
instead of 
```
    ...
    colorblind_safe: False
    compile: <CompilationOptions(
)>
```